### PR TITLE
[WIP] Handle get reference result 

### PR DESCRIPTION
**Pull request by krhubert**
dep https://github.com/mesg-foundation/engine/pull/1545

### CODE REVIEW

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -230,6 +233,20 @@ func (s *Orchestrator) resolveInput(wfHash hash.Hash, exec *execution.Execution,
 	return exec.Outputs.Fields[outputKey], nil
 }
 
+func (s *Orchestrator) resolveResult(wfHash hash.Hash, exec *execution.Execution, instanceHash hash.Hash, outputKey string) (*types.Value, error) {
+	if !wfHash.Equal(exec.ProcessHash) {
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: i'm not sure but i think `exec.ProcessHash` will never contains the same `wfHash` as it is created manually or by another process.

File: orchestrator/orchestrator.go [hunk 2]
```diff
@@ -210,6 +210,9 @@ func (s *Orchestrator) outputToValue(output *process.Process_Node_Map_Output, ou
 		if node.GetTask() != nil {
 			return s.resolveInput(wf.Hash, exec, v.Ref.NodeKey, outputKey)
 		}
+		if result := node.GetResult(); result != nil {
+			return s.resolveInput(wf.Hash, exec, v.Ref.NodeKey, outputKey)
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: should be `resolveResult` i guess

File: orchestrator/orchestrator.go [hunk 3]
```diff
@@ -230,6 +233,20 @@ func (s *Orchestrator) resolveInput(wfHash hash.Hash, exec *execution.Execution,
 	return exec.Outputs.Fields[outputKey], nil
 }
 
+func (s *Orchestrator) resolveResult(wfHash hash.Hash, exec *execution.Execution, instanceHash hash.Hash, outputKey string) (*types.Value, error) {
+	if !wfHash.Equal(exec.ProcessHash) {
+		return nil, fmt.Errorf("reference's result nodeKey not found")
+	}
+	if !exec.InstanceHash.Equal(instanceHash) {
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: i think this is not bad but you should also check that the `TaskKey` are the right one.

File: orchestrator/orchestrator.go (hunk 1 above)
Comment (krhubert) [reply orchestrator/orchestrator.go]: Can you add test about resolve result. It will help me a lot with implementing that logic and future refactor.

**Comment by NicolasMahe** [comment]
Closing this PR as the feature changed

**Pull request closed without merging**